fw/cache.c

```diff
@@ -1818,24 +1824,25 @@ tfw_cache_purge_invalidate(TfwHttpReq *req)
 /**
  * Process PURGE request method according to the configuration.
  */
-static void
+static int
 tfw_cache_purge_method(TfwHttpReq *req)
 {
 	int ret;
+	int res = -EINVAL;
 	TfwAddr saddr;
 	TfwGlobal *g_vhost = tfw_vhost_get_global();
 
 	/* Deny PURGE requests by default. */
-	if (!(cache_cfg.cache && g_vhost->cache_purge && g_vhost->cache_purge_acl)) {
-		tfw_http_send_resp(req, 403, "purge: not configured");
-		return;
-	}
+		if (!(cache_cfg.cache && g_vhost && g_vhost->cache_purge && g_vhost->cache_purge_acl)) {
+			tfw_http_send_resp(req, 403, "purge: not configured");
+			return res;
+		}
```

Comment:
bad identation

Reply:
done

Reply:
The usual Tempesta FW HTTP response for PURGE looks like
```
< HTTP/1.1 200 OK
< date: Sun, 26 Sep 2021 13:20:58 GMT
< content-length: 0
< server: Tempesta FW/0.7.0
```

and the underlying server (Nginx in my current settings) returns

```
< HTTP/1.1 200 OK
< Server: nginx/1.10.3
< Date: Sun, 26 Sep 2021 13:19:54 GMT
< Content-Type: text/html
< Content-Length: 3378
< Last-Modified: Thu, 02 Sep 2021 20:40:13 GMT
< Connection: keep-alive
< ETag: "613136ad-d32"
< Expires: Sun, 26 Sep 2021 13:29:54 GMT
< Cache-Control: max-age=600
< Cache-Control: public
< Accept-Ranges: bytes
<
```

I can remove or edit any headers but is it ok to leave all the rest? and how to analyse the headers? The returned headers set is dependent on upstream settings, there may be whatever list of headers.
I think about leaving Date, fixing Content-Length and Setting server header. and deleting the rest. But this all is barely load work than re-creating a short answer.